libretroshare/src/gxs/gxssecurity.cc

```diff
@@ -1110,19 +1110,28 @@ bool GxsSecurity::validateNxsGrp(const RsNxsGrp& grp, const RsTlvKeySignature& s
 		grpMeta.serialise(metaData, metaDataLen,api_versions_to_check[i]);
 
 		// copy msg data and meta in allmsgData buffer
-		memcpy(allGrpData, grp.grp.bin_data, grp.grp.bin_len);
-		memcpy(allGrpData+(grp.grp.bin_len), metaData, metaDataLen);
+		if(grp.grp.bin_data && grp.grp.bin_len>0)
```

Comment:
This change basically means that if the group has no data you do not verify the signature. That is just wrong. You need to put the if only around the memcpy of the grp.grp.bin_data.